src/v3/src/components/Widget/index.tsx

```diff
@@ -218,13 +220,21 @@ export const Widget: FunctionComponent<WidgetProps> = (widgetProps) => {
       return;
     }
 
-    if (pollingTransaction.nextStep?.name !== idxTransaction.nextStep?.name) {
+    /**
+     * polling transaction name can sometimes be the same as the current transaction's name
+     * i.e. unlocking an account and immeidiately being challenged with a different authenticator
+     * when that is the case, we need to compare inputs to determine if it is the same transaction
+     * lastly, inputs could also be the same (i.e. email otp to password) in that case we finally
+     * check the authenticator key to determine if they are the same
+     * (as you shouldnt be challenged with the same authenticator)
+    */
+    if (pollingTransaction.nextStep?.name !== idxTransaction.nextStep?.name
+        || !isEqual(pollingTransaction.nextStep?.inputs, idxTransaction.nextStep?.inputs)
+        || getAuthenticatorKey(pollingTransaction) !== getAuthenticatorKey(idxTransaction)) {
```

Comment:
nit suggestion: break the conditional out into a variable/variables for readability